clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java

```diff
@@ -6946,6 +6946,25 @@ public void testDescribeLogDirsPartialFailure() throws Exception {
             assertNotNull(result.descriptions().get(1).get());
         }
     }
+    
+    @Test
+    public void testDescribeLogDirsWithNonExistReplica() throws Exception {
+        int brokerId = 0;
+        TopicPartitionReplica tpr1 = new TopicPartitionReplica("topic1", 12, brokerId);
+        TopicPartitionReplica tpr2 = new TopicPartitionReplica("topic2", 12, brokerId);
+        try (AdminClientUnitTestEnv env = mockClientEnv()) {
+            env.kafkaClient().setNodeApiVersions(NodeApiVersions.create());
+            env.kafkaClient().prepareResponseFrom(
+                    new DescribeLogDirsResponse(
+                            new DescribeLogDirsResponseData().setResults(singletonList(
+                                    prepareDescribeLogDirsResult(tpr1, "/var/data/kafka0", 123456, 1, false)))),
+                    env.cluster().nodeById(brokerId));
+
+            DescribeReplicaLogDirsResult result = env.adminClient().describeReplicaLogDirs(asList(tpr1, tpr2));
+            Map<TopicPartitionReplica, KafkaFuture<DescribeReplicaLogDirsResult.ReplicaLogDirInfo>> values = result.values();
+            assertNull(values.get(tpr2).get().getCurrentReplicaLogDir());
```

Comment:
could you verify the result of `tpr1`?

Reply:
Yes, I will add assert tp1 currentReplicaLogDir